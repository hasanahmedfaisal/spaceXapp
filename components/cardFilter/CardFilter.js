import React from 'react'
import FilterButton from '../filterButton/FilterButton'
import range from 'lodash/range'
import isEmpty from 'lodash/isEmpty'

export default class CardFilter extends React.Component {
    render() {
        return (
            <div className="mainCardContainer">
                    <div className="filters">Filters</div>
                    <div className="launchYear">Launch Year</div>
                    <div className="buttonContainer">
                        {range(2006,2021).map((val) => (
                            <FilterButton name={val.toString()} 
                                filterType="launch_year"
                                selectedFilters={this.props.filters}
                                isSelected={!isEmpty(this.props.filters) && !!this.props.filters.launch_year && 
                                    this.props.filters.launch_year === val.toString()} />
                        ))}
                    </div>
                    <div className="launchYear">Successful Launch</div>
                    <div className="buttonContainer">
                        {["True", "False"].map(val => (
                            <FilterButton name={val} 
                                filterType="launch_success"
                                selectedFilters={this.props.filters}
                                isSelected={!isEmpty(this.props.filters) && !!this.props.filters.launch_success && 
                                    this.props.filters.launch_success === val} />
                        ))}
                    </div>
                    <div className="launchYear">Successful Landing</div>
                    <div className="buttonContainer">
                    {["True", "False"].map(val => (
                            <FilterButton name={val} 
                                filterType="land_success"
                                selectedFilters={this.props.filters}
                                isSelected={!isEmpty(this.props.filters) && !!this.props.filters.land_success && 
                                    this.props.filters.land_success === val} />
                        ))}
                    </div>
                </div>
        )
    }
}