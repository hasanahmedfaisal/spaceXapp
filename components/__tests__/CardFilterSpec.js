import CardFilter from '../cardFilter/CardFilter'
import renderer from 'react-test-renderer'

describe('Card Filter',() => {
    it('should render correctly',() => {
        const cardFilterProps = {
            filters: {
                "launch_year": '2010'
            }
        }
        const tree = renderer.create(<CardFilter {...cardFilterProps}/>).toJSON()
        expect(tree).toMatchSnapshot()
    }) 
})