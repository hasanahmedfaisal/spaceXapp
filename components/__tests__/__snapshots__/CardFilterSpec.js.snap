// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Filter should render correctly 1`] = `
<div
  className="mainCardContainer"
>
  <div
    className="filters"
  >
    Filters
  </div>
  <div
    className="launchYear"
  >
    Launch Year
  </div>
  <div
    className="buttonContainer"
  >
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2006
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2007
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2008
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2009
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button isSelected"
        onClick={[Function]}
      >
        2010
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2011
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2012
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2013
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2014
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2015
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2016
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2017
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2018
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2019
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        2020
      </button>
    </div>
  </div>
  <div
    className="launchYear"
  >
    Successful Launch
  </div>
  <div
    className="buttonContainer"
  >
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        True
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        False
      </button>
    </div>
  </div>
  <div
    className="launchYear"
  >
    Successful Landing
  </div>
  <div
    className="buttonContainer"
  >
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        True
      </button>
    </div>
    <div
      className="buttonDiv"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        False
      </button>
    </div>
  </div>
</div>
`;
